src/renderer/modals/finder/finder.jsx

```diff
@@ -0,0 +1,186 @@
+    if (event.code === 'Escape') {
+      this.props.close();
+    } else if (event.code === 'Enter') {
+      this.findNext();
+    }
+  }
+
+  render() {
+    return (
+      <div
+        id='finder'
+        onClick={this.props.focus}
+      >
+        <div className='finder'>
+          <div className='finder-input-wrapper'>
+            <input
+              className='finder-input'
+              placeholder=''
+              value={this.state.searchTxt}
+              onChange={this.searchTxt}
+              ref={(input) => {
+                this.searchInput = input;
+              }}
+            />
+            <span className={this.state.foundInPage ? 'finder-progress' : 'finder-progress finder-progress__disabled'}>{this.state.matches}</span>
+          </div>
+          <button
+            className='finder-prev'
+            onClick={this.findPrev}
+            disabled={!this.state.searchTxt}
+          >
+            <svg
+              xmlns='http://www.w3.org/2000/svg'
+              width='24'
+              height='24'
+              viewBox='0 0 24 24'
+              className='icon'
+              fill='none'
+              stroke='currentColor'
+              strokeWidth='2'
+              strokeLinecap='round'
+              strokeLinejoin='round'
+            >
+              <polyline points='18 15 12 9 6 15'/>
+            </svg>
+          </button>
+          <button
+            className='finder-next'
+            onClick={this.findNext}
+            disabled={!this.state.searchTxt}
+          >
+            <svg
+              xmlns='http://www.w3.org/2000/svg'
+              width='24'
+              height='24'
+              viewBox='0 0 24 24'
+              fill='none'
+              className='icon arrow-up'
+              stroke='currentColor'
+              strokeWidth='2'
+              strokeLinecap='round'
+              strokeLinejoin='round'
+            >
+              <polyline points='6 9 12 15 18 9'/>
+            </svg>
+          </button>
+          <button
+            className='finder-close'
+            onClick={this.props.close}
+          >
+            <svg
+              xmlns='http://www.w3.org/2000/svg'
+              width='24'
+              height='24'
+              viewBox='0 0 24 24'
+              fill='none'
+              className='icon'
+              stroke='currentColor'
+              strokeWidth='2'
+              strokeLinecap='round'
+              strokeLinejoin='round'
+            >
+              <line
+                x1='18'
+                y1='6'
+                x2='6'
+                y2='18'
+              />
+              <line
+                x1='6'
+                y1='6'
+                x2='18'
+                y2='18'
+              />
+            </svg>
+          </button>
+        </div>
+      </div>
+    );
+  }
+}
+
+Finder.propTypes = {
+  close: PropTypes.func,
+  findInPage: PropTypes.func,
+  stopFindInPage: PropTypes.func,
```

Comment:
It will always be passed in by its parent, the component cannot function correctly without it since you wouldn't be able to close the finder and clear the finder selection without it. It should throw an error in the console if it's not defined.

Reply:
I know, seems I started writting a comment about that, but it never got added (most likely my fault). I'll try to remember the gist of it:

so I'm wondering why do we pass those functions if there is nothing from the index interacting with it? might as well add all to the component and remove them from the props entirely.

Reply:
I guess I'm just following the pattern that we have set out for the modals and all other pieces in the renderer. We keep the external functions that communicate with the main process in the index file, along with the call to `ReactDOM.render` and the rest of the component functions normally.

There's no particular reason for this I don't think, but I would like to keep things consistent.